HID-1838: CD Header

rupl opened the pull request:
## https://humanitarian.atlassian.net/browse/HID-1838

Big ol' chunk of work to implement the Common Design Header. Included in the work were changes to the overall site layout and other components that were affected by the Header, such as the placement of the mobile back button.

- **CD Skip link:** we moved the link's position to match CD specs
- **CD Global Header:** OCHA Services, User menu, Language Switcher
- **CD Header:** Logo, Search, Main nav
- **CD Layout:** Removed the fluidity of the HID website to match CD specs (max-width, centered)
- **CD Footer:** keyboard nav matched to Header for consistency

The one major omission in this work was the top search bar, which we intentionally skipped because the CD search lacks autocomplete and other critical features that HID relies on. Rather than invent and implement them brand new, we left the HID search functionality as-is.

E2E coverage is vastly improved, but running the tests in series results in failures at specific tests. Also, some DB info might need to be adjusted, namely the users and passwords found in our [Trello card](https://trello.com/c/IOKlKYuZ/175-test-accounts).

rupl committed:
HID-1838: move skip-link out of header component

rupl committed:
HID-1838: paste CD markup into Header component

Full disclosure: translations, interactivity, assets, etc all broken.

rupl committed:
HID-1831: re-enable language E2E

rupl committed:
HID-1831: fix E2E for Search omnibar

rupl committed:
HID-1831: tabs -> spaces

rupl committed:
Merge remote-tracking branch 'origin/dev-migration' into hid/1838-cd-header

* origin/dev-migration:
  Bump eslint-utils from 1.3.1 to 1.4.2
  chore(package): update lockfile for karma-coverage
  chore(package): update karma-coverage to version 2.0.0
  HID-1817 - Upgrade google-libphonenumber

rupl committed:
Merge branch 'hid/1831-test-cleanup' into hid/1838-cd-header

* hid/1831-test-cleanup:
  HID-1831: tabs -> spaces
  HID-1831: fix E2E for Search omnibar
  HID-1831: re-enable language E2E

rupl committed:
HID-1838: tabs -> spaces

rupl committed:
HID-1838: tabs -> spaces

rupl committed:
HID-1838: restore OCHA Services menu

rupl committed:
HID-1838: E2E for OCHA Services

rupl committed:
HID-1838: copy customized links from HID production header

rupl committed:
HID-1838: restore CD Language Switcher

rupl committed:
HID-1838: restore main navigation in Header

rupl committed:
HID-1838: add all CD icons into assets dir

rupl committed:
HID-1838: only display search/nav when authenticated

rupl committed:
HID-1838: restore HID logo

rupl committed:
Merge remote-tracking branch 'origin/dev' into hid/1838-cd-header

* origin/dev:
  HID-1829 - Added instructions to separate redirect URLs by new lines
  HID-1829 - support multiple redirect URLs
  Bump mixin-deep from 1.3.1 to 1.3.2

rupl committed:
HID-1838: settle on layout for main search

rupl committed:
HID-1838: style search box to look like CD

rupl committed:
HID-1838: try to squish the main page layout into place

Naturally, the sidebar had to be adjusted

rupl committed:
HID-1838: user menu dropdown

rupl committed:
HID-1838: tabs -> spaces

rupl committed:
HID-1838: tabs -> spaces

rupl committed:
HID-1838: fix nav E2E tests

rupl committed:
HID-1838: fix anonymous login link on Global Header

rupl committed:
HID-1838: restore Notifications indicator

rupl committed:
HID-1838: fix regression in language switcher

rupl committed:
HID-1838: remove HTML-comment of old Header

rupl committed:
HID-1838: fix sidebar. again.

rupl committed:
HID-1838: fix E2E List suite

rupl committed:
HID-1838: fix E2E Language suite

rupl committed:
HID-1838: fix E2E Login suite

rupl committed:
HID-1838: more whitespace cleanup

rupl committed:
HID-1838: fix E2E Checkin/Checkout suite

rupl committed:
HID-1838: fix E2E Language suite when run in series

rupl committed:
Merge remote-tracking branch 'origin/dev-cd' into hid/1838-cd-header

* origin/dev-cd:
  Update all public-facing packages depending on set-value
  OPS-5922: cd config for new env
  HID-1829: bump package.json to match our git tags
  HID-1829: clarify instructions on new client form
  HID-1842 - skip suggestions typo fix
  Bump package.json to match git tags/releases
  Update all public-facing packages depending on set-value
  HID-1842 - fixed checking into a list but not subscribing to a service breaks functionality

rupl committed:
Add small delay to all login functions

rupl committed:
HID-1838: proper cleanup after running Language E2E suite

rupl committed:
HID-1838: nudge the Search autocomplete so it connects with search bar

rupl committed:
HID-1838: resolve error during Grunt workflow

rupl committed:
HID-1838: restore separator on logged-in user-menu

rupl committed:
HID-1838: add hover underline to user-menu/notifications

rupl committed:
HID-1838: lowercase the language indicator

rupl committed:
HID-1838: reorder AppController functions

No code changes, only reordering to place function definitions
nearer to each other when related, and also execute them after
each is defined.

rupl committed:
HID-1838: Global Header arrow icons should display dropdown state

rupl committed:
HID-1838: override CD colors to match HID branding

rupl committed:
HID-1838: CD feedback to pad sidebar differently

rupl committed:
HDI-1838: use same back button in all view modes

rupl committed:
HID-1838: avoid gray highlight on focused logo

rupl committed:
HID-1838: make more room for search at various breakpoints

rupl committed:
HID-1838: Language E2E should match CD specs

The language switcher should officially be in lowercase
and our old test matched previous HID behavior of uppercase.

rupl committed:
HID-1838: use standard commenting convention for HID overrides

rupl committed:
HID-1838: comprehensive keyboard nav styling for CD Header

rupl committed:
HID-1838: comprehensive keyboard nav styling for CD Footer

rupl committed:
HID-1838: match existing HID conventions for button focus in CD Header

rupl committed:
HID-1838: alow up/down keys to navigate CD Header dropdowns

guillaumev commented on src/app/components/client/ClientController.js:
```diff
@@ -23,8 +23,7 @@
       };
-      thisScope.client.redirectUrls = thisScope.client.redirectUrls.split("\n");
-      thisScope.client.redirectUrls = thisScope.client.redirectUrls.map(function (url) { return url.trim(); });
+      thisScope.client.redirectUrls = thisScope.client.redirectUrls.split("\n").map(function (url) { return url.trim(); });
```
@rupl Nothing to do with the CD right ;-) ?

rupl replied:
this is a merged-in change from `dev` branch. I guess I changed it somehow ¯\\\_(ツ)\_/¯

guillaumev replied:
no worries :-)

guillaumev commented on src/app/components/search/search-form.html:
```diff
@@ -45,3 +56,26 @@
 
 </div>
+
+<!--
```
@rupl Would it make sense to completely remove this, since it seems commented out ? Or are you keeping it in there for a specific reason ?

rupl replied:
yes totally.

Since these migrations took a long time I had worked with both copies next to each other for some time. If you see any other commented blocks of HTML, please do call them out and I'll remove all of them.

guillaumev commented:
@rupl Besides the code commented out which should be removed, it looks fine to me. I pulled it on my local and it seems to work fine, so good to go I believe after you remove the code that's been commented out.

rupl committed:
HDI-1838: remove some old HTML comments

rupl committed:
HID-1838: audit all HID overrides of CD and mark them thoroughly

In case we ever want to upgrade CD on this site, having everything
clearly commented will produce a more readable diff that will
hopefully help resolve any merge conflicts.
